aws/src/test/java/org/apache/iceberg/aws/TestHttpClientConfigurations.java

```diff
@@ -0,0 +1,404 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.aws;
+
+import java.time.Duration;
+import java.util.Map;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+import org.junit.Assert;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mockito;
+import software.amazon.awssdk.http.apache.ApacheHttpClient;
+import software.amazon.awssdk.http.urlconnection.UrlConnectionHttpClient;
+
+public class TestHttpClientConfigurations {
+  @Test
+  public void testUrlConnectionConnectionSocketTimeoutConfiguration() {
+    Map<String, String> properties = Maps.newHashMap();
+    properties.put(AwsProperties.HTTP_CLIENT_URLCONNECTION_SOCKET_TIMEOUT_MS, "90");
+    properties.put(AwsProperties.HTTP_CLIENT_URLCONNECTION_CONNECTION_TIMEOUT_MS, "80");
+    AwsProperties awsProperties = new AwsProperties(properties);
+    UrlConnectionHttpClientConfigurations urlConnectionHttpClientConfigurations =
+        new UrlConnectionHttpClientConfigurations();
+    urlConnectionHttpClientConfigurations.initialize(
+        awsProperties.urlConnectionHttpClientProperties());
+    UrlConnectionHttpClient.Builder urlConnectionHttpClientBuilder =
+        UrlConnectionHttpClient.builder();
+    UrlConnectionHttpClient.Builder spyUrlConnectionHttpClientBuilder =
+        Mockito.spy(urlConnectionHttpClientBuilder);
+    ArgumentCaptor<Duration> socketTimeoutCaptor = ArgumentCaptor.forClass(Duration.class);
+    ArgumentCaptor<Duration> connectionTimeoutCaptor = ArgumentCaptor.forClass(Duration.class);
+
+    urlConnectionHttpClientConfigurations.configureUrlConnectionHttpClientBuilder(
+        spyUrlConnectionHttpClientBuilder);
+    Mockito.verify(spyUrlConnectionHttpClientBuilder).socketTimeout(socketTimeoutCaptor.capture());
+    Mockito.verify(spyUrlConnectionHttpClientBuilder)
+        .connectionTimeout(connectionTimeoutCaptor.capture());
+
+    Duration capturedSocketTimeout = socketTimeoutCaptor.getValue();
+    Duration capturedConnectionTimeout = connectionTimeoutCaptor.getValue();
+
+    Assert.assertEquals(
+        "The configured socket timeout should be 90 ms", 90, capturedSocketTimeout.toMillis());
+    Assert.assertEquals(
+        "The configured connection timeout should be 80 ms",
+        80,
+        capturedConnectionTimeout.toMillis());
+  }
+
+  @Test
+  public void testUrlConnectionConnectionTimeoutConfiguration() {
```

Comment:
why do we need this test? isn't it a subset of the test above?

Reply:
ok. I see it is just refactored over from `TestAwsProperties`. still find it a little weird to test `override 2 configs` vs `override 1 config`. anyway, this is fine by me.

Reply:
I think the purpose is to verify that the default behavior is to not configure `socketTimeout` if corresponding property is not set:
```
Mockito.verify(spyUrlConnectionHttpClientBuilder, Mockito.never())
        .socketTimeout(socketTimeoutCaptor.capture());
```

I agree that these tests are somewhat redundant. I plan to refactor them to have two tests testing default behavior of all configs for `UrlConnectionHttpClient` and `ApacheHttpClient`.